server/sandbox_stop_freebsd.go

```diff
@@ -31,51 +30,21 @@ func (s *Server) stopPodSandbox(ctx context.Context, sb *sandbox.Sandbox) error
 		return nil
 	}
 
-	// Get high-performance runtime hook to trigger preStop step for each container
-	hooks, err := runtimehandlerhooks.GetRuntimeHandlerHooks(ctx, &s.config, sb.RuntimeHandler(), sb.Annotations())
-	if err != nil {
-		return fmt.Errorf("failed to get runtime handler %q hooks", sb.RuntimeHandler())
-	}
-
-	podInfraContainer := sb.InfraContainer()
-	containers := sb.Containers().List()
-	containers = append(containers, podInfraContainer)
-
-	const maxWorkers = 128
-	var waitGroup errgroup.Group
-	for i := 0; i < len(containers); i += maxWorkers {
-		max := i + maxWorkers
-		if len(containers) < max {
-			max = len(containers)
-		}
-		for _, ctr := range containers[i:max] {
-			cStatus := ctr.State()
-			if cStatus.Status != oci.ContainerStateStopped {
-				if ctr.ID() == podInfraContainer.ID() {
-					continue
-				}
-				c := ctr
-				waitGroup.Go(func() error {
-					if err := s.stopContainer(ctx, c, stopTimeoutFromContext(ctx)); err != nil {
-						return fmt.Errorf("failed to stop container for pod sandbox %s: %v", sb.ID(), err)
-					}
-					if err := s.nri.stopContainer(ctx, sb, c); err != nil {
-						return err
-					}
-					return nil
-				})
-			}
-			if hooks != nil {
-				if err := hooks.PreStop(ctx, ctr, sb); err != nil {
-					log.Warnf(ctx, "Failed to run PreStop hook for container %s in pod sandbox %s: %v", ctr.Name(), sb.ID(), err)
-				}
```

Comment:
note to reviewers/future self: this is done in https://github.com/saschagrunert/cri-o/blob/97b02aad5d83d06e60855bdf71b2e4c8f4e5fb21/server/container_stop.go#L43-L77